capstone/src/main/java/com/google/sps/gmail/GmailAPILoader.java

```diff
@@ -0,0 +1,98 @@
+package com.google.sps.gmail;
+
+import com.google.api.client.auth.oauth2.Credential;
+import com.google.api.client.googleapis.auth.oauth2.GoogleCredential;
+import com.google.api.client.googleapis.javanet.GoogleNetHttpTransport;
+import com.google.api.client.http.javanet.NetHttpTransport;
+import com.google.api.client.json.JsonFactory;
+import com.google.api.client.json.jackson2.JacksonFactory;
+import com.google.api.services.gmail.Gmail;
+import com.google.common.collect.Streams;
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.net.URLEncoder;
+import java.security.GeneralSecurityException;
+import java.util.LinkedHashMap;
+import java.util.Map;
+import java.util.stream.Collectors;
+import org.json.JSONObject;
+
+public class GmailAPILoader {
+  private static final JsonFactory JSON_FACTORY = JacksonFactory.getDefaultInstance();
```

Comment:
I saw some code where a comment said that best practice is to have a single global instance of json factory and datastore factory and the like. Maybe we can do that?